src/main/java/com/minecraftabnormals/abnormals_core/core/api/model/FullbrightModelLoader.java

```diff
@@ -0,0 +1,259 @@
+import java.util.function.Function;
+
+public class FullbrightModelLoader implements ISimpleModelGeometry<FullbrightModelLoader> {
+    public static final Gson GSON = (new GsonBuilder())
+            .registerTypeAdapter(FullbrightBlockPart.class, new FullbrightBlockPart.Deserializer())
+            .registerTypeAdapter(FullbrightBlockPartFace.class, new FullbrightBlockPartFace.Deserializer())
+            .registerTypeAdapter(BlockFaceUV.class, new BlockFaceUV.Deserializer())
+            .create();
+
+    private final List<FullbrightBlockPart> elements;
+
+    public FullbrightModelLoader(List<FullbrightBlockPart> list) {
+        this.elements = list;
+    }
+
+    @Override
+    public void addQuads(IModelConfiguration owner, IModelBuilder<?> modelBuilder, ModelBakery bakery, Function<RenderMaterial, TextureAtlasSprite> spriteGetter, IModelTransform modelTransform, ResourceLocation modelLocation) {
+        for (FullbrightBlockPart part : this.elements) {
+            for (Direction direction : part.mapFaces.keySet()) {
+                FullbrightBlockPartFace face = (FullbrightBlockPartFace) part.mapFaces.get(direction);
+                TextureAtlasSprite sprite = spriteGetter.apply(owner.resolveTexture(face.texture));
+
+                BakedQuad quad = BlockModel.makeBakedQuad(part, face, sprite, direction, modelTransform, modelLocation);
+                if ((!face.override && part.fullbright) || (face.override && face.fullbright))
+                    LightUtil.setLightData(quad, 0xF000F0);
+
+                if (face.cullFace == null) {
+                    modelBuilder.addGeneralQuad(quad);
+                } else {
+                    modelBuilder.addFaceQuad(modelTransform.getRotation().rotateTransform(face.cullFace), quad);
+                }
+            }
+        }
+    }
+
+    @Override
+    public Collection<RenderMaterial> getTextures(IModelConfiguration owner, Function<ResourceLocation, IUnbakedModel> modelGetter, Set<Pair<String, String>> missingTextureErrors) {
+        Set<RenderMaterial> textures = Sets.newHashSet();
+        for (FullbrightBlockPart part : this.elements) {
+            for (BlockPartFace face : part.mapFaces.values()) {
+                textures.add(owner.resolveTexture(face.texture));
+            }
+        }
+
+        return textures;
+    }
+
+    public static class Loader implements IModelLoader<FullbrightModelLoader> {
+        public static final FullbrightModelLoader.Loader INSTANCE = new FullbrightModelLoader.Loader();
+
+        private Loader() {}
+
+        @Override
+        public void onResourceManagerReload(IResourceManager manager) {}
+
+        @Override
+        public FullbrightModelLoader read(JsonDeserializationContext context, JsonObject model) {
+            List<FullbrightBlockPart> list = this.getModelElements(model);
+            return new FullbrightModelLoader(list);
+        }
+
+        private List<FullbrightBlockPart> getModelElements(JsonObject model) {
```

Comment:
Does this really need to be a separate method?
It's fairly small and is only used in one other method.